### ISSUE TITLE
Laptop's external screen goes to standby every 30 seconds

### ISSUE THREAD

**Issue by jhakonen** [pop-os/pop#831]
**Distribution (run `cat /etc/os-release`):**
```
$ cat /etc/os-release
NAME="Pop!_OS"
VERSION="19.10"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 19.10"
VERSION_ID="19.10"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=eoan
UBUNTU_CODENAME=eoan
LOGO=distributor-logo-pop-os
```
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
```
$ LANG=C apt policy gnome-settings-daemon
gnome-settings-daemon:
  Installed: 3.34.1-1ubuntu1pop0~1571080815~19.10~d7a3d94
  Candidate: 3.34.1-1ubuntu1pop0~1571080815~19.10~d7a3d94
  Version table:
 *** 3.34.1-1ubuntu1pop0~1571080815~19.10~d7a3d94 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu eoan/main amd64 Packages
        100 /var/lib/dpkg/status
     3.34.1-1ubuntu1 500
        500 http://us.archive.ubuntu.com/ubuntu eoan/main amd64 Packages
```

**Issue/Bug Description:**
My laptop's external screen goes to standby after 30 seconds of idle period. This is a some what rare occurrence, as it usually takes a few days for this issue to pop up. But when it occurs it keeps on happening constantly until I do one of the following:
* Restart the laptop,
* Lock the screen (I've used Super+L) and then unlock it, or
* Run: `xset -dpms`
The final option seems to just hide it, as running `xset +dpms` causes the issue to reoccur.
It is as if the screen blank time which is 15 minutes by default is set to 30 seconds, which is less than you can even set from Gnome's power settings.

I followed what happened in dbus when the problem occurred with command:
```
$ dbus-monitor --session
```
Whenever the screen went to standby there was method call which set "PowerSaveMode" to zero on path /org/gnome/Mutter/DisplayConfig. The sender was :1.96. With Following commands I got the sender's process:
```
$ gdbus call --session --dest org.freedesktop.DBus --object-path / --method org.freedesktop.DBus.GetConnectionUnixProcessID :1.96
(uint32 3052,)
$ ps aux | grep 3052
jhakonen  3052  0.0  0.1 362104 30496 ?        Ssl  11:34   0:02 /usr/lib/gnome-settings-daemon/gsd-power
...
```
I then added `--verbose` to the daemon's arguments in /usr/lib/systemd/user/gsd-power.service to get more detailed logging.

This was a few days ago.

Today the problem reoccurred and I took logs of what happened ([see attached gsd-power-bug.log](https://github.com/pop-os/pop/files/4081791/gsd-power-bug.log)):
```
$ journalctl --user-unit=gsd-power.service > ~/gsd-power-bug.log
```
21:34 is when I first noticed that the screen went to standby way too fast. And it happened again at 21:36. I then fixed the problem by locking the screen with Super+L and then unlocked it, this happened at 21:42.
In the log file there's an interesting error a few minutes before the screen goes to standby on first time:
```
tammi 18 21:32:04 pop-os gsd-power[3052]: Error setting property 'PowerSaveMode' on interface org.gnome.Mutter.DisplayConfig: GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying (g-dbus-error-quark, 4)
```
I'm not sure how to continue debugging the issue.

**Steps to reproduce (if you know):**
No idea how to reproduce this.

**Expected behaviour:**
The screen should go to standby after the 15 minutes (as configured in Gnome's settings), not after 30 seconds.

**Other Notes:**
My laptop is MSI GS65 Stealth Thin 8RF.
The external screen (ASUS PG348Q) is connected to laptop through Lenovo's Thinkpad TB3 dock.

I didn't have this issue with Ubuntu 19.04. The problem started occurring in Pop OS 19.10. I upgraded to Pop OS 19.10 by installing Pop OS from scratch, but kept my /home partition intact.

**Comment by MrMinemeet** [comment]
Seem to have the same issue on Pop!_os 20.04 LTS with my Razer Blade 15 (2020)

**Comment by fat-tire** [comment]
Seeing this issue in Ubuntu 23.04 (beta) weirdly.  Traced it similar to above via dbus to `/usr/libexec/gsd-power` which is started at the similar `/usr/lib/systemd/user/org.gnome.SettingsDaemon.Power.service` file.

I tried killing that process, and it immediately restarted the `gsd-power`.  Only suddenly somehow it doesn't shut off the screen after 30 seconds this time as it did before.  Dunno if it will survive a reset/relogin, but perhaps just killing it and letting it restart is a workaround?  Should I find something more permanent, I'll report here and maybe I'll try adding `--verbose` and seeing if I can figure out what is causing it.

**Comment by colans** [comment]
Maybe mark this one as a duplicate of https://github.com/pop-os/pop/issues/2104 as there's more info & discussion there?

**Comment by leviport** [comment]
I believe you're correct, since `xset -dpms` fixes it, just like users have mentioned in #2104 . I'm going to close this as a duplicate. Thanks for pointing that out.
